app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -157,70 +173,62 @@ export default class PendingTransactionTracker extends EventEmitter {
 
       return
     }
-    // *note to self* hard failure point
-    const transactionReceipt = await this.query.getTransactionReceipt(txHash)
-
-
-    // If another tx with the same nonce is mined, set as dropped.
-    const taken = await this._checkIfNonceIsTaken(txMeta)
-    let dropped
-    try {
-      // check the network if the nonce is ahead the tx
-      // and the tx has not been mined into a block
-      dropped = await this._checkIfTxWasDropped(txMeta, transactionReceipt)
-
-      // the dropped buffer is in case we ask a node for the tx
-      // that is behind the node we asked for tx count
-      // IS A SECURITY FOR HITTING NODES IN INFURA THAT COULD GO OUT
-      // OF SYNC.
-      // on the next block event it will return fire as dropped
-      if (typeof this.droppedBuffer[txHash] !== 'number') {
-        this.droppedBuffer[txHash] = 0
-      }
-
-      // 3 block count buffer
-      if (dropped && this.droppedBuffer[txHash] < 3) {
-        dropped = false
-        ++this.droppedBuffer[txHash]
-      }
-
-      if (dropped && this.droppedBuffer[txHash] === 3) {
-        // clean up
-        delete this.droppedBuffer[txHash]
-      }
-    } catch (e) {
-      log.error(e)
-    }
 
-    if (taken || dropped) {
+    if (await this._checkIfNonceIsTaken(txMeta)) {
       this.emit('tx:dropped', txId)
       return
     }
 
-    // get latest transaction status
-    if (transactionReceipt?.blockNumber) {
-      this.emit('tx:confirmed', txId, transactionReceipt)
-    } else {
-      const err = new Error('Missing transaction receipt or block number.')
+    try {
+      const transactionReceipt = await this.query.getTransactionReceipt(txHash)
+      if (transactionReceipt?.blockNumber) {
+        this.emit('tx:confirmed', txId, transactionReceipt)
+        return
+      }
+    } catch {
```

Comment:
We should capture the real error here - it would be more useful than "Missing transaction receipt". The stack trace would tell us that we're missing a transaction receipt.